redis/src/cluster_async/mod.rs

```diff
@@ -446,56 +450,89 @@ where
 
     // Query a node to discover slot-> master mappings.
     fn refresh_slots(&mut self) -> impl Future<Output = RedisResult<()>> {
+        self.refresh_slots_with_retries(None)
+    }
+    // Query a node to discover slot-> master mappings with retries
+    fn refresh_slots_with_retries(
+        &mut self,
+        retries: Option<Arc<atomic::AtomicUsize>>,
+    ) -> impl Future<Output = RedisResult<()>> {
+        const MAX_REQUESTED_NODES: usize = 50;
         let inner = self.inner.clone();
 
         async move {
-            let mut write_guard = inner.conn_lock.write().await;
-            let mut connections = mem::take(&mut write_guard.0);
-            let slots = &mut write_guard.1;
-            let mut result = Ok(());
-            for (_, conn) in connections.iter_mut() {
-                let mut conn = conn.clone().await;
-                let value = match conn.req_packed_command(&slot_cmd()).await {
-                    Ok(value) => value,
-                    Err(err) => {
-                        result = Err(err);
-                        continue;
-                    }
-                };
-                match parse_slots(value, inner.cluster_params.tls).and_then(|v| {
-                    Self::build_slot_map(slots, v, inner.cluster_params.read_from_replicas)
-                }) {
+            let read_guard = inner.conn_lock.write().await;
+            let num_of_nodes = read_guard.0.len();
+            let amount = std::cmp::min(num_of_nodes, MAX_REQUESTED_NODES);
+            let mut requested_nodes = {
+                let mut rng = thread_rng();
+                read_guard.0.values().choose_multiple(&mut rng, amount)
+            };
+            let mut cluster_slot_futures = Vec::new();
+            for conn in requested_nodes.iter_mut() {
+                let mut conn: C = conn.clone().await;
+                let slots_future = async move { conn.req_packed_command(&slot_cmd()).await };
+                #[cfg(feature = "tokio-comp")]
+                cluster_slot_futures.push(tokio::spawn(slots_future));
```

Comment:
what do you get by spawning instead of just calling join_all on the iterator itself?